tutorial/markdown/java/spring-boot.md

```diff
@@ -53,172 +95,364 @@ mvn package
+  "callsign": "Couchbase",
+  "iata": "CB",
+  "icao": "CBA",
+  "country": "United States",
+  "active": true
 }
 ```
 
-As we can see, we want our user's password to be encrypted in the database too, we can achieve this simply with bcrypt, a dependency we have installed.
+The `id` field is the unique identifier for the document. The `type` field is used to identify the type of document. The `name` field is the name of the airline. The `callsign` field is the callsign of the airline. The `iata` field is the IATA code of the airline. The `icao` field is the ICAO code of the airline. The `country` field is the country of the airline. The `active` field is a boolean value indicating whether the airline is active or not.
+
+### Code Organization
+
+- `src/test/java`: Contains integration tests.
+- `src/main/java/org/couchbase/quickstart/springboot/repositories`: Contains the repository implementation.
+- `src/main/java/org/couchbase/quickstart/springboot/models`: Contains the data model.
+- `src/main/java/org/couchbase/quickstart/springboot/controllers`: Contains the RESTful API controllers.
+- `src/main/java/org/couchbase/quickstart/springboot/services`: Contains the service classes.
+
+## Data Model
+
+For this tutorial, we use three collections, `airport`, `airline` and `route` that contain sample airports, airlines and airline routes respectively. The route collection connects the airports and airlines as seen in the figure below. We use these connections in the quickstart to generate airports that are directly connected and airlines connecting to a destination airport. Note that these are just examples to highlight how you can use SQL++ queries to join the collections.
+
+![Data Model](./travel_sample_data_model.png)
 
 ## Let's Review the Code
 
 To begin clone the repo and open it up in the IDE of your choice to learn about how to create, read, update and delete documents in your Couchbase Server.
 
-## POST a Profile
+### Integration Tests
 
-For CRUD operations we will use the [Key Value operations](https://docs.couchbase.com/java-sdk/current/howtos/kv-operations.html) that are built into the Couchbase SDK to create, read, update, and delete a document. Every document will need an ID (similar to a primary key in other databases) in order to save it to the database.
+`AirlineIntegrationTest.java`
+This class contains integration tests for the application. It tests various functionalities like getting, creating, updating, and deleting airlines, as well as listing airlines by country and destination airport.
 
-Open the ProfileController file found in the controllers folder and navigate to the save method.  Let’s break this code down. First, in the save method signature we have a ProfileRequest object. This brings in the information passed to the controller in the body of the request and is used to get a profile variable from the ProfileRequest object's getProfile method. Note that we auto-generate a random UUID for the pid.
+### Repository
```

Comment:
If you mention the repository, model & service, you should mention at least one sentence about the pattern with a link to learn more.

Reply:
done